oml/models/siamese.py

```diff
@@ -45,37 +50,107 @@ def forward(self, x1: Tensor, x2: Tensor) -> Tensor:
         return y
 
 
-class ResNetSiamese(IPairwiseModel):
+class ConcatSiamese(IPairwiseModel, IFreezable):
     """
-    The model takes two images as inputs, passes them through backbone and
-    estimates the distance between them.
+    This model concatenates two inputs and passes them through
+    a given backbone and applyies a head after that.
 
     """
 
-    def __init__(self, pretrained: bool) -> None:
+    pretrained_models: Dict[str, Any] = {}
+
+    def __init__(
+        self,
+        extractor: IExtractor,
+        mlp_hidden_dims: List[int],
+        weights: Optional[Union[str, Path]] = None,
+        strict_load: bool = True,
+    ) -> None:
         """
         Args:
-            pretrained: Set ``True`` if you want to use pretrained model.
+            extractor: Instance of ``IExtractor`` (e.g. ``ViTExtractor``)
+            mlp_hidden_dims: Hidden dimensions of the head
+            weights: Path to weights file or ``None`` for random initialization
+            strict_load: Whether to use ``self.load_state_dict`` with strict argument
 
         """
-        super(ResNetSiamese, self).__init__()
-        self.backbone = resnet18(pretrained=pretrained)
+        super(ConcatSiamese, self).__init__()
+        self.extractor = extractor
+
+        self.head = MLP(
+            in_channels=self.extractor.feat_dim,
+            hidden_channels=[*mlp_hidden_dims, 1],
+            activation_layer=Sigmoid,
+            dropout=0.5,
```

Comment:
Maybe it makes sense to put the `dropout` parameter to the correspondent yaml?

Reply:
@dapladoc hank you for the review! I guess after a series of experiments we return to the model and rework it, including parameters handling